### ISSUE TITLE
Increase Paste/Fetch character limit

### ISSUE THREAD

**Issue by hepcat72** [galaxyproject/galaxy#5349]
Hi,

Our facility has a high throughput sequencing web interface that manages our sequencing requests, and the way we currently handle transfers of data from there to galaxy (for users) is that their demultiplexed samples each get gzipped independently and a series of links to the files is created.  The user then copies these links, login to their galaxy account, and use the Paste/fetch upload tool to retrieve the data directly from our sequencing system for loading into their galaxy account.

Unfortunately, the paste/fetch text box, while it allows the full list of links to be pasted without visibly truncating it, somewhere along the line truncates the series of links somewhere inside the 6th link, which creates an empty 6th file.  The first 5 files are successfully uploaded and the remainder do not transfer.  So the user must paste the links 5 at a time, opening a new paste/fetch text box for every 5 links.

I don't know whether this is a bug or whether this interface was not designed to handle multiple long links the way we're using it?  I requested those who handle the sequencing interface to create shorter links, but they've got other issues that are higher priority.  It's admittedly only a minor inconvenience, but at the same time, it seems like it might be an easy fix.  Or perhaps there's some other means we can provide to the user for transferring their files from our other system in a single step?  At the least, if there's a character limit to the paste/fetch box, it could be helpful to not show all the pasted characters so that the user has an earlier realization that what they're doing is not going to work.

Thanks,
Rob

**Comment by martenson** [comment]
@hepcat72 we have touched the fetchbox recently to address truncation there, what version is your Galaxy?

**Comment by hepcat72** [comment]
17.09

**Comment by martenson** [comment]
@hepcat72 could you please provide an example of what is being pasted (with line endings and everything)?

**Comment by hepcat72** [comment]
No problem.  I just did it myself this morning.  I will edit the links (security via obscurity) but I'll keep them the same length:

```
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-0000-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-0000-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-0000-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-1111-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-1111-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-1111-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-222-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-222-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-222-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-333-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-333-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/AA-333-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-0000-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-0000-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-0000-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-1111-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-1111-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-1111-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-222-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-222-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-222-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-333-A_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-333-B_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/BB-333-C_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz
https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/unmatched_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz <https://myurl.universit.edu/tmp/XXXXXXXXXXXXXXXXX/unmatched_Demultiplexed-fastq-fileset-sample-Username-24RNA-1801-from-flowcell-YYYYYYYYY-on-2018-01-18_Read_1_passed_filter.fastq.gz>
```

Rob

**Comment by martenson** [comment]
@hepcat72  I replicated the issue, we will fix this quickly, thanks for reporting it!

**Comment by hepcat72** [comment]
Cool.  Thanks.  While you're in there, it would also be nice if the names of the files in galaxy didn't include the entire URL from the "https://..." to the last "/".  I mean, I'd still have to edit the names, but that would be one less click/drag/delete action I'd have to do manually for each sample.

Rob

**Comment by martenson** [comment]
@hepcat72 we have some history regarding this (https://github.com/galaxyproject/galaxy/issues/4005 and https://github.com/galaxyproject/galaxy/issues/4162) tldr: it is not a perfect solution to expect that the filenames after the last `/` are unique and we did not come up with something we were happy with yet.

**Comment by hepcat72** [comment]
Yeah, that thought occurred to me.  I suppose a quick mitigation of that issue would be to only chop off the URL if they all pass a uniqueness test first, otherwise use the whole URL with a warning about file name uniqueness.

Rob

**Comment by martenson** [comment]
@hepcat72 this issue will be fixed with #5353 - it is rather silly to be frank, sorry about that and thanks for reporting
